### PR TITLE
feat(core): add new option for home page

### PR DESCRIPTION
**Pull request by stonebuzz**
### Changes description

Add new option for home page
- Only forms
- Forms and list of request

image
 

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

Closes #N/A

### CODE REVIEW

File: front/wizard.php [hunk 1]
```diff
@@ -50,8 +50,31 @@
...
+   echo "<br/>";
+
+   //backup session value
+   $save_session_fold_search = $_SESSION['glpifold_search'];
+   //hide search if need
+   if (PluginFormcreatorEntityconfig::getUsedConfig('is_search_issue_visible', Session::getActiveEntity()) == PluginFormcreatorEntityconfig::CONFIG_SEARCH_ISSUE_HIDDEN) {
+      $_SESSION['glpifold_search'] = true;
+   }
+
+   Search::show('PluginFormcreatorIssue');
```
Comment (btry) [inline front/wizard.php:73]:
```suggestion
   Search::show(PluginFormcreatorIssue::class);
```

File: css/styles.scss [hunk 2]
```diff
@@ -622,6 +622,12 @@ tr[data-itemtype="PluginFormcreatorCondition"] ~ tr[data-itemtype="PluginFormcre
    }
 }
 
+#plugin_formcreator_wizard_categories .merge_with_request {
```
Comment (btry) [inline css/styles.scss:625]:
```suggestion
#plugin_formcreator_wizard_categories .plugin_formcreator_merge_with_request {
```

File: inc/form.class.php [hunk 3]
```diff
@@ -583,14 +583,19 @@ public function showList() : void {
...
       echo '</div>';
    }
 
-   public function showWizard() : void {
-      echo '<div id="plugin_formcreator_wizard_categories" class="card">';
+   public function showWizard(bool $with_request_list = false) : void {
+
+      $class = "";
+      if ($with_request_list){
+         $class = "merge_with_request";
```
Comment (btry) [inline inc/form.class.php:596]:
```suggestion
         $class = "plugin_formcreator_merge_with_request";
```

File: install/upgrade_to_2.14.php [hunk 4]
```diff
@@ -43,8 +43,26 @@ public function upgrade(Migration $migration) {
...
+
+      if (!$DB->fieldExists($table, 'home_page')) {
+         $this->migration->addField($table, 'home_page', 'integer', ['after' => 'tile_design', 'value' => '-2']);
+         $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `home_page`= 1 WHERE `entities_id` = 0");
+      }
+
+      if (!$DB->fieldExists($table, 'tile_design')) {
+         $this->migration->addField($table, 'tile_design', 'integer', ['after' => 'home_page', 'value' => '-2']);
+         $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `tile_design`= '0' WHERE `entities_id` = 0");
+      }
```
Comment (btry) [inline install/upgrade_to_2.14.php:62]: this column already exists, these lines should be removed

File: inc/entityconfig.class.php [hunk 5]
```diff
@@ -158,6 +161,14 @@ public static function getEnumUIForm() : array {
       ];
    }
 
+   public static function getEnumHomePage() : array {
+      return [
+         self::CONFIG_PARENT                       => __('Inheritance of the parent entity'),
+         self::CONFIG_HOME_PAGE_ONLY_FORM          => __('Only default forms', 'formcreator'),
+         self::CONFIG_HOME_PAGE_FORM_AND_REQUEST   => __('Forms and list of requests', 'formcreator'),
+      ];
+   }
```
Comment (btry) [inline inc/entityconfig.class.php:170]: need a unit tests for this method

File: front/wizard.php (hunk 1 above)
Comment (stonebuzz) [reply front/wizard.php]: done !

File: css/styles.scss (hunk 2 above)
Comment (stonebuzz) [reply css/styles.scss]: done !

File: install/upgrade_to_2.14.php (hunk 4 above)
Comment (stonebuzz) [reply install/upgrade_to_2.14.php]: done !

File: inc/entityconfig.class.php (hunk 5 above)
Comment (stonebuzz) [reply inc/entityconfig.class.php:170]: done!

File: front/issue.php [hunk 6]
```diff
@@ -54,17 +54,7 @@
...
-//hide search if need
-if (PluginFormcreatorEntityconfig::getUsedConfig('is_search_issue_visible', Session::getActiveEntity()) == PluginFormcreatorEntityconfig::CONFIG_SEARCH_ISSUE_HIDDEN) {
-   $_SESSION['glpifold_search'] = true;
-}
-
-Search::show('PluginFormcreatorIssue');
-
-//restore session value
-$_SESSION['glpifold_search'] = $save_session_fold_search;
+PluginFormcreatorCommon::showGenericSearchIssueIfNeeded();
```
Comment (btry) [inline front/issue.php:57]: I would avoid such name for a method. 

```suggestion
PluginFormcreatorCommon::showGenericSearchIssue();
```

File: inc/common.class.php [hunk 7]
```diff
@@ -842,4 +842,18 @@ public static function showMiniDashboard(): void {
          echo "</div>";
       }
    }
+
+   public static function showGenericSearchIssueIfNeeded(): void {
```
Comment (btry) [inline inc/common.class.php:846]:
```suggestion
   public static function showGenericSearchIssue(): void {
```

File: inc/form.class.php [hunk 8]
```diff
@@ -584,13 +584,29 @@ public function showList() : void {
...
+
       echo '<div id="plugin_formcreator_wizard" class="card-group">';
-      $this->showWizard();
+      $this->showWizard($need_request_list);
       echo '</div>';
+
+      if ($need_request_list) {
+         echo "<br/>";
+         echo "<br/>";
+         PluginFormcreatorCommon::showGenericSearchIssueIfNeeded();
```
Comment (btry) [inline inc/form.class.php:599]:
```suggestion
         PluginFormcreatorCommon::showGenericSearchIssue();
```

File: inc/form.class.php [hunk 9]
```diff
@@ -637,6 +653,7 @@ public function showWizard() : void {
       echo Html::scriptblock("$(function() {
          plugin_formcreator.updateWizardFormsView();
       });");
+
```
Comment (btry) [inline inc/form.class.php:656]:
```suggestion
```

File: install/upgrade_to_2.14.php [hunk 10]
```diff
@@ -43,6 +43,17 @@ public function upgrade(Migration $migration) {
...
+   }
+
+   public function addEntityOption() {
+      global $DB;
+      $table = 'glpi_plugin_formcreator_entityconfigs';
+
+      if (!$DB->fieldExists($table, 'home_page')) {
+         $this->migration->addField($table, 'home_page', 'integer', ['after' => 'tile_design', 'value' => '-2']);
+         $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `home_page`= 1 WHERE `entities_id` = 0");
+      }
```
Comment (btry) [inline install/upgrade_to_2.14.php:56]: Migration class provides a way to update a field after it has been added. This allows to remove the if() making code a bit more readable. See Migration::addField()

Might a good idea to unit test it in tests/1-install/Config.php (along other entity settings by the way)

```suggestion
      $this->migration->addField(
         $table, 
         'home_page', 
         'integer', [
            'after' => 'tile_design', 
            'value' => '-2',
            'update' => '1',
            'condition' => 'WHERE `entities_id` = 0'
         ]
      );
```

File: tests/3-unit/PluginFormcreatorEntityConfig.php [hunk 11]
```diff
@@ -118,6 +121,7 @@ public function testGetUsedConfig() {
          'replace_helpdesk' => \PluginFormcreatorEntityconfig::CONFIG_PARENT,
          'is_kb_separated'  => \PluginFormcreatorEntityconfig::CONFIG_PARENT,
          'sort_order'       => \PluginFormcreatorEntityconfig::CONFIG_PARENT,
+         'homepage'       => \PluginFormcreatorEntityconfig::CONFIG_PARENT,
```
Comment (btry) [inline tests/3-unit/PluginFormcreatorEntityConfig.php:124]:
```suggestion
         'homepage'         => \PluginFormcreatorEntityconfig::CONFIG_PARENT,
```

**Review by btry — APPROVED** [review]

**Pull request merged**